### ISSUE TITLE
fingerprinting protection should be done in C++ instead of js content script

### ISSUE THREAD

**Issue by diracdeltas** [brave/browser-laptop#12045]
follow-up to https://github.com/brave/browser-laptop/pull/11784

as @snyderp showed above, blocking DOM methods in a JS content script is a flawed approach because there are sneaky ways to get references to the methods. it would be better to disable the DOM methods in chromium/muon.

cc @bridiver

**Comment by SergeyZhukovsky** [comment]
That how it is done on Android browser in C++. I think we can use most on it https://github.com/brave/browser-android-tabs/commit/c6fdea38de6f818346109e82c23308acb93cee01

**Comment by snyderp** [comment]
Its also how the Tor folks do their Canvas fingerprint blocking (which, last I checked, is "prompt user with permission dialog whenever `toDataUrl` is called").

I have more experience with the Tor/Mozilla codebase at the moment, but I think it should not be too difficult to get something similar pushed into chromium/muon.  I don't know where this is on the timeline for you all, but happy to help if outside help would be welcome

**Comment by ghostwords** [comment]
If there are gaps in the WebExtensions APIs, situations where extensions are unable to guarantee before-anything-else execution (such as parent frames being able to access child frames before child frame content scripts had a chance to do their thing), I think these are browser bugs to be fixed. If this is what it is, we should file crbug and/or Mozilla@Bugzilla issues.

**Comment by snyderp** [comment]
I agree completely.  I'll open issues with both teams in the next 24 hours and post links here to follow up.

**Comment by snyderp** [comment]
Bugs below:

Firefox: https://bugzilla.mozilla.org/show_bug.cgi?id=1424176
Chrome: https://bugs.chromium.org/p/chromium/issues/detail?id=793217

**Comment by diracdeltas** [comment]
assigning to @jumde for now

**Comment by diracdeltas** [comment]
the fix for this issue would also fix https://github.com/brave/browser-laptop/issues/12110 and https://github.com/brave/browser-laptop/issues/11683

**Comment by bsclifton** [comment]
Moving to 0.21.x

**Comment by alexwykoff** [comment]
@diracdeltas is this release blocking?

**Comment by diracdeltas** [comment]
non-blocking, it can be moved to one release later

**Comment by bsclifton** [comment]
Confirmed fixed with brave-core 😄
